src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetPackageSearchService.cs

```diff
@@ -326,6 +328,37 @@ private async ValueTask<IInstalledAndTransitivePackages> GetInstalledAndTransiti
             }
         }
 
+        /// <summary>
+        /// Combines package Folders from PackageReferenceProject with global package folders
+        /// </summary>
+        /// <param name="projectContextInfos">A collection of projects</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        private async Task<IReadOnlyList<SourceRepository>> GetAllPackageFoldersAsync(
+            IReadOnlyCollection<IProjectContextInfo> projectContextInfos,
+            CancellationToken cancellationToken)
+        {
+            Task<IReadOnlyCollection<string>>[] tasks = projectContextInfos.Select(pctxi => pctxi.GetPackageFoldersAsync(_serviceBroker, cancellationToken).AsTask()).ToArray();
+
+            IReadOnlyCollection<string>[] packageFolders = await Task.WhenAll(tasks);
+
+            HashSet<string> pkgFoldersUnique = new HashSet<string>();
+
+            packageFolders.ForEach(folders => pkgFoldersUnique.AddRange(folders));
+
+            IEnumerable<SourceRepository> assetsPackageFolders = pkgFoldersUnique.Select(folder => _sharedServiceState.SourceRepositoryProvider.CreateRepository(new PackageSource(folder)));
+
+            IEnumerable<SourceRepository> globalPackageFolderRepositories = await _globalPackageFolderRepositoriesLazy.GetValueAsync(cancellationToken);
+
```

Comment:
That's a lot of blank lines. One blank line per line of code seems excessive. Can we group some lines of code and only put a blank line when we want to logically separate concepts in the code to make it easier to read?